File: CSES/1. Introductory Problems/CSES-1755_Palindrome Reorder.py
S = input()

cnt = [0] * 26
for ch in S:
    cnt[ord(ch) - ord('A')] += 1
flag = True
ans = ""
mid = ""
for i in range(26):
    ch = chr(i + ord('A'))
    if cnt[i] & 1:
        if mid != "":
            flag = False
            break
        mid = ch
    ans += ch * (cnt[i] // 2)
if not flag:
    print("NO SOLUTION")
else:
    print(ans + mid + ans[::-1])

File: CSES/1. Introductory Problems/CSES-1755_Palindrome Reorder.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    string s;
    cin >> s;
    int cnt[26] = {0};
    for (char ch : s) {
        cnt[ch - 'A']++;
    }
    bool flag = true;
    string ans = "", mid = "";
    for (int i = 0; i < 26; i++) {
        char ch = i + 'A';
        if (cnt[i] & 1) {
            if (mid != "") {
                flag = false;
                break;
            }
            mid = ch;
        }
        ans += string(cnt[i] / 2, ch);
    }
    if (!flag) {
        cout << "NO SOLUTION" << endl;
    } else {
        cout << ans + mid + string(ans.rbegin(), ans.rend()) << endl;
    }
    return 0;
}